src/data/transforms/normalize.rs

```diff
@@ -0,0 +1,164 @@
+//! The Normalizing Transformer
+//!
+//! This module contains the `Normalizer` transformer.
+//!
+//! The `Normalizer` transformer is used to transform input data
+//! so that the l2 norm of each rows are all equal to 1.
+//! If input data has a row with all 0, `Normalizer` keeps the row as it is.
+//!
+//! Because transformation is performed per row independently,
+//! inverse transformation is not supported.
+//!
+//! # Examples
+//!
+//! ```
+//! use rusty_machine::data::transforms::{Transformer, Normalizer};
+//! use rusty_machine::linalg::Matrix;
+//!
+//! // Constructs a new `Normalizer`
+//! let mut transformer = Normalizer::default();
+//!
+//! let inputs = Matrix::new(2, 2, vec![-1.0, 2.0, 1.5, 3.0]);
+//!
+//! // Transform the inputs
+//! let transformed = transformer.transform(inputs).unwrap();
+//! ```
+
+use learning::error::{Error, ErrorKind};
+use linalg::{Matrix, BaseMatrix, BaseMatrixMut};
+use super::Transformer;
+
+use libnum::Float;
+
+/// The Normalizer
+///
+/// The Normalizer provides an implementation of `Transformer`
+/// which allows us to transform the all rows to have the same l2 norm (1).
+///
+/// See the module description for more information.
+#[derive(Debug)]
+pub struct Normalizer;
+
+/// Create a `Normalizer`.
+impl Default for Normalizer {
```

Comment:
This can be derived on the above struct (though I don't think even this is necessary).

```rust
let a = Normalizer;
let b = Normalizer::default();
let c = Normalizer::new();
```

These all do the same thing here. I'd suggest we only keep the first.